plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt

```diff
@@ -1,14 +1,38 @@
 package org.jlleitschuh.gradle.ktlint
 
+import java.io.File
 import java.io.PrintWriter
 import javax.inject.Inject
+import org.gradle.api.file.FileType
 import org.gradle.api.model.ObjectFactory
 import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.TaskAction
+import org.gradle.work.ChangeType
+import org.gradle.work.InputChanges
 
 @Suppress("UnstableApiUsage")
 @CacheableTask
 open class KtlintCheckTask @Inject constructor(
     objectFactory: ObjectFactory
 ) : BaseKtlintCheckTask(objectFactory) {
     override fun additionalConfig(): (PrintWriter) -> Unit = {}
+
+    @TaskAction
+    fun lint(inputChanges: InputChanges) {
+        project.logger.info("Executing ${if (inputChanges.isIncremental) "incrementally" else "non-incrementally"}")
+
+        val filesToLint = inputChanges
+            .getFileChanges(stableSources)
+            .fold(mutableSetOf<File>()) { acc, fileChange ->
+                logger.debug("File changed: $fileChange")
+                with(fileChange) {
+                    if (fileType != FileType.DIRECTORY &&
+                        changeType != ChangeType.REMOVED) {
+                        acc.add(file)
+                    }
+                }
+                acc
+            }
```

Comment:
You could also do `map` with a `filterNonNull` afterward. This works too though.

Reply:
Force updated this part.